### PR TITLE
[EXC-7258] cookies & vanity domains

### PR DESCRIPTION
**Pull request by hiuxmaycry**

### CODE REVIEW

**Comment by hiuxmaycry** [comment]
I created a draft for the customer facing documentation the cookie consent managent feature. Please make a review of this @wwarner-salesforce.

**Comment by wwarner-salesforce** [comment]
Excellent-- thank you @hiuxmaycry ! I will review this PR and add to it as needed.

File: modules/ROOT/pages/portal-vanity-domain.adoc [hunk 1]
```diff
@@ -119,6 +119,32 @@ NOTE: Your server must run using `https`.
 
 If set up correctly, you can access and navigate to `+https://developers.robocop.com/path/of/justice/+` exactly as if you were at `+https://anypoint.mulesoft.com/exchange/portals/robo-cop/+`.
 
+== Set up Cookie Consent Management for your domain
+
+A cookie consent managment solution is provided out of the box for Exchange public portals that don't use vanity domains.
```
Comment (jyotirmi) [inline modules/ROOT/pages/portal-vanity-domain.adoc]: Spell check: management
Also, we should perhaps add a link to what is cookie consent management. That document does not exist yet.

**Review by jyotirmi — COMMENTED** [review]
Left a comment

**Comment by wwarner-salesforce** [comment]
@hiuxmaycry https://www.mulesoft.org/jira/browse/EXC-7258 says the customer must provide the organization ID and the domain. The draft here says the customer must provide the organization domain and the domain. Is `robo-cop` the "organization ID" or the "organization domain"?

File: modules/ROOT/pages/portal-vanity-domain.adoc (hunk 1 above)
Comment (hiuxmaycry) [reply modules/ROOT/pages/portal-vanity-domain.adoc]: Hi @jyotirmi, I do not know if there will be any external page that we can add as a reference or if one will be created soon within the mulesoft docs, in any case I could add an example image of the consent manager cookie that we currently have in exchange, which is a modal like this:

image

This is not an explanation but the customer could get an idea of what it is 😛 .

**Comment by hiuxmaycry** [comment]
> @hiuxmaycry https://www.mulesoft.org/jira/browse/EXC-7258 says the customer must provide the organization ID and the domain. The draft here says the customer must provide the organization domain and the domain. Is `robo-cop` the "organization ID" or the "organization domain"?

Hi @wwarner-salesforce the value needed is the "organization domain", my bad, I updated the ticket with organization domain.

**Comment by wwarner-salesforce** [comment]
@hiuxmaycry Perfect. Thank you!

File: modules/ROOT/pages/cookies.adoc [hunk 2]
```diff
@@ -0,0 +1,158 @@
+= Configure Cookies
```
Comment (jyotirmi) [inline modules/ROOT/pages/cookies.adoc]: @electronwill - Suggested heading - About Exchange Cookies

Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc]: Sounds good. I've updated the PR to use "About Exchange Cookies".

File: modules/ROOT/pages/cookies.adoc [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+endif::[]
+:imagesdir: ../assets/images/
+
+Anypoint Exchange uses cookies to save user settings, enhance user experience, and improve performance.
+
+Exchange requires some cookies to work correctly. These include cookies for session details and authentication information.
+
+Other functional cookies are not required, although certain functionality does not work without them. These include cookies for site analytics and advertising.
+
+Exchange provides public portal website visitors with cookie consent management and full documentation of all cookies, in compliance with marketing and privacy regulations throughout the world.
```
Comment (jyotirmi) [inline modules/ROOT/pages/cookies.adoc]: remove "throughout the world"

File: modules/ROOT/pages/cookies.adoc [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images/
+
+Anypoint Exchange uses cookies to save user settings, enhance user experience, and improve performance.
+
+Exchange requires some cookies to work correctly. These include cookies for session details and authentication information.
+
+Other functional cookies are not required, although certain functionality does not work without them. These include cookies for site analytics and advertising.
```
Comment (jyotirmi) [inline modules/ROOT/pages/cookies.adoc]: Functional cookies are not needed for the website to work correctly. It's only the advertising cookies, which we think may interfere with some functionality such as videos.

**Review by jyotirmi — COMMENTED** [review]
See my comments embedded

File: modules/ROOT/pages/cookies.adoc (hunk 3 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc]: Removed.

File: modules/ROOT/pages/cookies.adoc (hunk 4 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc]: I updated the text and corrected the information about the functional cookies.

**Review by mcampo — DISMISSED** [review]

File: modules/ROOT/pages/cookies.adoc [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+The cookie consent manager shows settings for required cookies, functional cookies, and advertising cookies, and general information about cookies.
+
+Website visitors use the cookie consent manager to view the cookies the site uses, view descriptions of all individual cookies and understand what these cookies are doing, and enable or disable functional and advertising cookies.
+
+== Vanity Domains
+
+If you enable a vanity domain for your Exchange public portal, enable cookie consent management with the xref:portal-vanity-domain.adoc[public portal vanity domain instructions].
+
+== Cookies Used by Exchange
```
Comment (wwarner-salesforce) [inline modules/ROOT/pages/cookies.adoc:40]: @mcampo I wanted to check this table with you.

In https://www.mulesoft.org/jira/browse/DOCS-9632 there was a link to https://docs.google.com/spreadsheets/d/1c9DDosW3kFdfR12vHdDwbb5I57lbuiw1NEeOw6VM3es/edit#gid=1373564656 and I tried to get the cookie info from there. I did see that there is another spreadsheet at https://docs.google.com/spreadsheets/d/1CBC0pVJoNBBaacQRQXM3c7DHIPvzOUVndUvkAfFnd1Q/ , which appears to be older.

Am I using the right spreadsheet? Is the table missing any info, and is there any incorrect info? What should I enter as the description for the cookies that currently have no description?

Thank you!

File: modules/ROOT/pages/cookies.adoc [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+|ajs_anonymous_id |365 days |Funuctional |Counts how many people visit the site for site analytics.
+|ajs_group_id |365 days |Functional |Tracks visitor usage and events within the website.
+|ajs_user_id |365 days |Functional |Helps track visitor usage, events, and target marketing, and sometimes also measures application performance and stability.
+|amplitude_cookie_test |Session |Functional |
+|amplitude_id_7856322d613393b800727439f0954eb9mulesoft.com |3650 days |Functional |
+|amplitude_idundefinedmulesoft.com |Session |Functional |
+|amplitude_testmulesoft.com |Session |Functional |
+|exchange.sess |Session |Required |Maintains a user session within Exchange. No personal information is stored in the cookie.
+|exchange.sess.sig |Session |Required |Stores the signature used to validate the Exchange session cookie.
+|IDE |729 days |Functional |Supports the Doubleclick platform, a real time bidding advertising exchange.
```
Comment (mcampo) [inline modules/ROOT/pages/cookies.adoc]:
```suggestion
|IDE |729 days |Advertising |Supports the Doubleclick platform, a real time bidding advertising exchange.
```

File: modules/ROOT/pages/cookies.adoc [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+|JSESSIONID |Session |Functional |Supports the New Relic platform for monitoring performance of web and mobile applications.
+|mulesoft.sess |Session |Required |Maintains a user session within Anypoint Platform. No personal information is stored in the cookie.
+|mulesoft.sess.sig |Session |Required |Stores the signature used to validate the Anypoint Platform session cookie.
+|test_cookie |Session |Functional |Supports the Doubleclick platform, a real time bidding advertising exchange.
```
Comment (mcampo) [inline modules/ROOT/pages/cookies.adoc]:
```suggestion
|test_cookie |Session |Advertising |Supports the Doubleclick platform, a real time bidding advertising exchange.
```

**Review by mcampo — COMMENTED** [review]
Corrected categorization of 2 cookies

File: modules/ROOT/pages/cookies.adoc (hunk 5 above)
Comment (mcampo) [reply modules/ROOT/pages/cookies.adoc:40]: @wwarner-salesforce https://docs.google.com/spreadsheets/d/1c9DDosW3kFdfR12vHdDwbb5I57lbuiw1NEeOw6VM3es/edit#gid=1373564656 is the one we should use 👍

Comment (mcampo) [reply modules/ROOT/pages/cookies.adoc:40]: @hiuxmaycry could you try to get info about the cookies that don't have a description?

File: modules/ROOT/pages/cookies.adoc [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+
+If you enable a vanity domain for your Exchange public portal, enable cookie consent management with the xref:portal-vanity-domain.adoc[public portal vanity domain instructions].
+
+== Cookies Used by Exchange
+
+[cols="1,1,1,5",options=header,width="100%",frame="topbot"]
+|===
+|Cookie Name |Duration |Cookie type |Description
+|__cfduid |Session |Required |Helps Cloudflare detect malicious visitors to our customers' websites and minimizes blocking legitimate users.
+|+__tld__+ |Session |Functional |
```
Comment (hiuxmaycry) [inline modules/ROOT/pages/cookies.adoc]: `__tld__` description: "Used to track visitors on multiple websites, in order to present relevant advertisement based on the visitor's preferences."

File: modules/ROOT/pages/cookies.adoc [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+|_csrf |Session |Required |Helps prevent Cross-Site Request Forgery attacks for site security.
+|_ga |730 days |Functional |Calculates visitor, session, and campaign data for site analytics.
+|_gat |Session |Functional |Throttles the request rate to limit the collection of data on high traffic sites.
+|_gid |1 day |Functional |
```
Comment (hiuxmaycry) [inline modules/ROOT/pages/cookies.adoc]: `_gid` Description: "Registers a unique ID that is used to generate statistical data on how the visitor uses the website."

File: modules/ROOT/pages/cookies.adoc [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+|ajs%3Acookies |365 days |Functional |Supports use, service performance, and functionality of websites.
+|ajs%3Atest |365 days |Functional |Counts visits and tracks other website traffic-related metrics for performance analytics.
+|ajs_anonymous_id |365 days |Funuctional |Counts how many people visit the site for site analytics.
+|ajs_group_id |365 days |Functional |Tracks visitor usage and events within the website.
+|ajs_user_id |365 days |Functional |Helps track visitor usage, events, and target marketing, and sometimes also measures application performance and stability.
+|amplitude_cookie_test |Session |Functional |
```
Comment (hiuxmaycry) [inline modules/ROOT/pages/cookies.adoc]: `amplitude_cookie_test` Description: "Detects whether partner data synchronization is functioning and currently running."

File: modules/ROOT/pages/cookies.adoc [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+|amplitude_id_7856322d613393b800727439f0954eb9mulesoft.com |3650 days |Functional |
+|amplitude_idundefinedmulesoft.com |Session |Functional |
+|amplitude_testmulesoft.com |Session |Functional |
```
Comment (hiuxmaycry) [inline modules/ROOT/pages/cookies.adoc]: `amplitude_id_7856322d613393b800727439f0954eb9mulesoft` `amplitude_idundefinedmulesoft.com` `amplitude_testmulesoft.com` description for all: "We use Amplitude to identify unique users visits in order to keep accurate web analytics."

Not found the real description for this cookies in Amplitude documentation.

File: modules/ROOT/pages/cookies.adoc (hunk 5 above)
Comment (hiuxmaycry) [reply modules/ROOT/pages/cookies.adoc:40]: Added description for those cookies 🙂

Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc:40]: That's perfect, and I believe the table is ready now. Thanks to you both! Please re-approve if this looks good to you. @hiuxmaycry @mcampo

**Review by mcampo — DISMISSED** [review]
👍

File: modules/ROOT/pages/cookies.adoc [hunk 12]
```diff
@@ -0,0 +1,64 @@
+= About Exchange Cookies
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images/
+
+Anypoint Exchange uses cookies to save user settings, enhance user experience, and improve performance.
```
Comment (Mulesoft-Mysti) [inline modules/ROOT/pages/cookies.adoc]: enhance "the" user experience (?)

File: modules/ROOT/pages/cookies.adoc [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+
+Exchange normally uses some functional cookies, including cookies for site analytics. Exchange does not require any functional cookies for site functionality.
+
+Content in Exchange documentation often uses advertising cookies. If a video embedded in an Exchange portal is hosted on a third-party site such as YouTube that uses advertising cookies, then the video is not visible to a website visitor who disables advertising cookies, and the visitor sees a message explaining this.
+
+== Cookie Consent Management
+
+Exchange provides public portal website visitors with cookie consent management and full documentation of all cookies, in compliance with marketing and privacy regulations.
+
+Some countries require websites not to use any non-essential cookie until after the website visitor has granted explicit consent.
```
Comment (Mulesoft-Mysti) [inline modules/ROOT/pages/cookies.adoc]: Would this be easier to read if written in a positive way? Like "Some countries require that websites use non-essential cookies only after the website visitor..."

File: modules/ROOT/pages/cookies.adoc [hunk 14]
```diff
@@ -0,0 +1,64 @@
+= About Exchange Cookies
+ifndef::env-site,env-github[]
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images/
+
+Anypoint Exchange uses cookies to save user settings, enhance user experience, and improve performance.
+
+In general, websites have required cookies necessary for the site to work correctly, functional cookies necessary for certain site functionality, functional cookies that are not necessary for any site functionality, and advertising cookies.
```
Comment (Mulesoft-Mysti) [inline modules/ROOT/pages/cookies.adoc]: would this be easier to read as a bullet list? (easier to see the contrast?) Like:

In general, websites use different types of cookies:

*  Necessary...
*  Functional...
* Functional...
* Advertising...

**Review by Mulesoft-Mysti — DISMISSED** [review]

File: modules/ROOT/pages/cookies.adoc (hunk 12 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc]: Yes, I think that is better. Updated. Thank you!

File: modules/ROOT/pages/cookies.adoc (hunk 13 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc]: I agree, that is better. Updated. Thank you!

File: modules/ROOT/pages/cookies.adoc (hunk 14 above)
Comment (wwarner-salesforce) [reply modules/ROOT/pages/cookies.adoc]: I like the bullet list idea. Updated. Thanks!

**Review by Mulesoft-Mysti — APPROVED** [review]

**Pull request merged**